modules/firehose/main.tf

```diff
@@ -141,12 +141,84 @@ resource "aws_iam_role_policy" "firehose_to_http_metric_policy" {
            "Resource": [
                "${aws_cloudwatch_log_group.firehose_loggroup.arn}"
            ]
+        },
+        {
+          "Effect": "Allow",
+          "Action": [
+              "lambda:InvokeFunction",
+              "lambda:GetFunctionConfiguration"
+          ],
+          "Resource": "${aws_lambda_function.lambda_processor.arn}:*"
         }
     ]
 }
 EOF
 }
 
+data "aws_iam_policy_document" "lambda_assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "lambda_iam" {
+  name               = "cx-cw-metrics-tags-lambda-iam"
+  assume_role_policy = data.aws_iam_policy_document.lambda_assume_role.json
+}
+
+resource "aws_iam_role_policy" "lambda_iam_policy" {
+  name   = "cx-cw-metrics-tags-lambda-iam-policy"
+  role   = aws_iam_role.lambda_iam.id
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+      {
+          "Action": [
+              "tag:GetResources",
+              "ec2:DescribeTransitGateway*",
+              "ec2:DescribeTags",
+              "ec2:DescribeRegions",
+              "ec2:DescribeInstances",
+              "dms:DescribeReplicationTasks",
+              "dms:DescribeReplicationInstances",
+              "apigateway:GET"
+          ],
+          "Effect": "Allow",
+          "Resource": "*",
+          "Sid": ""
+      },
+      {
+          "Action": [
+              "logs:PutLogEvents",
+              "logs:CreateLogStream",
+              "logs:CreateLogGroup"
+          ],
+          "Effect": "Allow",
+          "Resource": "arn:aws:logs:*:*:*",
+          "Sid": ""
+      }
+  ]
+}
+EOF
+}
+
+resource "aws_lambda_function" "lambda_processor" {
+  filename      = "path/to/function.zip" # TODO: Change to official Coralogix S3 bucket
```

Comment:
So I think we'll want to change this section to correspond to S3 bucket / object. So to make it work with every region, we should have something like:
```
s3_bucket="cx-cw-metrics-tags-lambda-processor-${aws_region}"
s3_key="function.zip"
```